subprojects/core/src/main/java/org/gradle/cache/internal/WrapperDistributionCleanupAction.java

```diff
@@ -17,45 +17,145 @@
 package org.gradle.cache.internal;
 
 import com.google.common.annotations.VisibleForTesting;
-import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Sets;
 import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.filefilter.RegexFileFilter;
 import org.gradle.api.Action;
-import org.gradle.api.UncheckedIOException;
-import org.gradle.api.logging.Logger;
-import org.gradle.api.logging.Logging;
 import org.gradle.util.GradleVersion;
 
 import java.io.File;
+import java.io.FileFilter;
 import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import static org.apache.commons.io.filefilter.FileFilterUtils.directoryFileFilter;
+import static org.gradle.util.CollectionUtils.single;
 
 public class WrapperDistributionCleanupAction implements Action<GradleVersion> {
 
     @VisibleForTesting static final String WRAPPER_DISTRIBUTION_FILE_PATH = "wrapper/dists";
-    private static final Logger LOGGER = Logging.getLogger(WrapperDistributionCleanupAction.class);
-    private static final ImmutableList<String> DISTRIBUTION_TYPES = ImmutableList.of("bin", "all");
+    private static final Pattern DISTRIBUTION_PATTERN = Pattern.compile("^gradle-(\\d.+)-(?:(all)|(bin))$");
```

Comment:
Do we need any pattern at all? I think we can handle all distributions using the same approach.

Reply:
I kept it for performance reasons but it's probably a premature optimization.

Reply:
Yeah I think so too